src/main.lua

```diff
@@ -1775,13 +1765,26 @@ end, function(msg)
 
 	print("Ticking from " .. lastCreatedEpochIndex .. " to " .. targetCurrentEpochIndex)
 
+	-- tick the demand factor
+	local demandFactor = demand.updateDemandFactor(msg.Timestamp)
+	if demandFactor ~= nil then
+		table.insert(newDemandFactors, demandFactor)
```

Comment:
Will this work correctly in the case where multiple epochs were ticked?

Reply:
demand factor has no affiliation to epochs - it has it's own period length that happens to be the same. `updateDemandFactor` handles that iteration based on the timestamp, so it'll iterate through periods to the current timestamp accordingly.